test/utils/runners.go

```diff
@@ -1220,15 +1220,17 @@ func DoPrepareNode(client clientset.Interface, node *v1.Node, strategy PrepareNo
 func DoCleanupNode(client clientset.Interface, nodeName string, strategy PrepareNodeStrategy) error {
 	var err error
 	for attempt := 0; attempt < retries; attempt++ {
-		node, err := client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+		var node *v1.Node
+		node, err = client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
 		if err != nil {
 			return fmt.Errorf("Skipping cleanup of Node: failed to get Node %v: %v", nodeName, err)
 		}
 		updatedNode := strategy.CleanupNode(node)
 		if apiequality.Semantic.DeepEqual(node, updatedNode) {
 			return nil
 		}
-		if _, err = client.CoreV1().Nodes().Update(context.TODO(), updatedNode, metav1.UpdateOptions{}); err == nil {
+		_, err = client.CoreV1().Nodes().Update(context.TODO(), updatedNode, metav1.UpdateOptions{})
+		if err == nil {
```

Comment:
Hi(こんにちは!) @oomichi -san! Thanks for the checking.

From [this wording](https://github.com/kubernetes/kubernetes/blob/master/test/utils/runners.go#L1240), we can expect that [this area](https://github.com/kubernetes/kubernetes/blob/master/test/utils/runners.go#L1222-L1238) is coded to retry a specific number of times in case of a conflict.

The most important part of this PR is [here](https://github.com/kubernetes/kubernetes/pull/101853/files#diff-6111c679b8726d3d6624c71f2ab9c954dd6d0ee95bb5ac3ca058b84abe328c16R1223) and the following playgrounds will give you an idea of what this PR want to say
https://play.golang.org/p/qCAMseQAfIy

---

And the following change was made because I personally feel it makes the scope of err easier to understand, but it's not essential, so I'll revert.
https://github.com/kubernetes/kubernetes/pull/101853/files#diff-6111c679b8726d3d6624c71f2ab9c954dd6d0ee95bb5ac3ca058b84abe328c16R1232

Reply:
In the old logic, multiple values are returned and assigned to new and old variables. In this case, the inner `err` is always new, and has nothing to do with the outer `err`, and hence the outer `err` cannot get assigned the proper value.

This PR ensures `err` always holds the same pointer, and hence be assigned properly after `retries`.